src/tools/weather/openMeteo.ts

```diff
@@ -169,10 +169,14 @@ export class OpenMeteoTool extends Tool<OpenMeteoToolOutput, ToolOptions, ToolRu
       const start = startDate
         ? new Date(startDate)
         : new Date(Date.UTC(now.getFullYear(), now.getMonth(), now.getDate()));
-      const end = endDate
+      let end = endDate
         ? new Date(endDate)
         : new Date(Date.UTC(now.getFullYear(), now.getMonth(), now.getDate()));
 
+      if (end < start) {
+        end = new Date(start);
+      }
+
```

Comment:
Good catch; however, we should rather throw an error and let the LLM regenerate.

```ts
throw new ToolInputValidationError(`The value of the 'end_date' property (${endDate}) must be greater than the value of 'start_date' (${startDate}).`);
```

What do you think?

Reply:
- If the llm provided an end date before the start date, then we should throw an error. 
- if the llm only provides a future start date, and we set an end data to satisfy the service which happens to be prior to the start date, then we should make sure that does not happen.

Setting the end date to start date + N will cause the service to provide weather data for the N days between start and end date which is probably not ideal. Ill fix the code to handle the 2 cases above.